### PR TITLE
[4.0] Routing: Fix notice in NoMenu rule

### PR DESCRIPTION
**Pull request by Hackwar**
Pull Request for Issue #25837.

### Summary of Changes
This is a fix based on the report of an issue in J3 and a subsequent codereview. I can't replicate the original posters issue, but when the issue is fixed by the PR #34049, then this one would have to be merged as well.

### Testing Instructions
So about those testing instructions... See above text.

### CODE REVIEW

File: libraries/src/Component/Router/Rules/NomenuRules.php [hunk 1]
```diff
@@ -139,7 +139,8 @@ public function build(&$query, &$segments)
 		{
 			$item = $this->router->menu->getItem($query['Itemid']);
 
-			if (!isset($query['option']) || ($item && $item->query['option'] === $query['option']))
+			if (!isset($query['option'])
+				|| ($item && isset($query['option']) && $item->query['option'] === $query['option']))
```
Comment (richard67) [inline libraries/src/Component/Router/Rules/NomenuRules.php]: See https://github.com/joomla/joomla-cms/pull/34049#discussion_r636335580 .

Comment (richard67) [inline libraries/src/Component/Router/Rules/NomenuRules.php]:
```suggestion
				|| ($item && isset($item->query['option']) && $item->query['option'] === $query['option']))
```

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on 3ab2470ebfb03666d4437f1ab6383a326e207dfb<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34050">issues.joomla.org/tracker/joomla-cms/34050</a>.</sub>

**Comment by richard67** [comment]
I have tested this item :white_check_mark: successfully on 3ab2470ebfb03666d4437f1ab6383a326e207dfb<br /><br />Code review.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34050">issues.joomla.org/tracker/joomla-cms/34050</a>.</sub>

**Comment by richard67** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/34050">issues.joomla.org/tracker/joomla-cms/34050</a>.</sub>

**Comment by richard67** [comment]
@wilsonge As this PR is identical with #34049 for staging, I leave it up to you if you want to merge this one here or if you want to merge the other one up.

**Pull request merged**